### PR TITLE
Add nexits average

### PR DESCRIPTION
**Pull request by Luni-4**
Average `NExit` over the total number of functions/closures in a space.

Thanks in advance for your review! :)

### CODE REVIEW

**Comment by marco-c** [comment]
This needs to be rebased

**Comment by Luni-4** [comment]
Done

File: src/metrics/exit.rs [hunk 1]
```diff
@@ -121,3 +140,27 @@ impl Exit for JavaCode {}
...
 impl Exit for CssCode {}
 impl Exit for HtmlCode {}
+
+#[cfg(test)]
+mod tests {
+    use std::path::PathBuf;
+
+    use super::*;
+
+    #[test]
```
Comment (marco-c) [inline src/metrics/exit.rs:150]: Can you also add a "basic" test with a single function? And one with two functions which are not nested?

File: src/metrics/exit.rs [hunk 2]
```diff
@@ -121,3 +140,27 @@ impl Exit for JavaCode {}
...
+            "def f(a, b):
+                 def foo(a):
+                     if a:
+                         return 1
+                 bar = lambda a: lambda b: b or True or True
+                 return bar(foo(a))(a)",
+            "foo.py",
+            PythonParser,
+            nexits,
+            [(exit, 2, usize)],
```
Comment (marco-c) [inline src/metrics/exit.rs:162]: Shouldn't there be 4 exits here in total (one for `f`, one for `foo`, one for each lambda)?

Comment (Luni-4) [reply src/metrics/exit.rs:162]: We don't consider closures exit points for now. Perhaps we should add them in another PR

Comment (marco-c) [reply src/metrics/exit.rs:193]: Can you file an issue about that, so we don't forget?

**Review by marco-c — APPROVED** [review]

File: src/metrics/exit.rs (hunk 2 above)
Comment (marco-c) [reply src/metrics/exit.rs:193]: https://github.com/mozilla/rust-code-analysis/issues/341

**Pull request merged**